### PR TITLE
Add tests in AddressBook, LogicManager and ModelManager

### LINKED ISSUE #260: [PE-D] Unexpected list of student for view
Build Number: Version Number 1.3
Reported By: Chu Yi-Ting
Reported On: 2020-10-30
Reason: Defect
Severity: Low
Environment: macOS version 10.15.7

Description:
view -s [student name] should show the students whose name match the “student name” according to the user guide.  However, the real implementation shows all the students with partial match.

Expected Behavior: 
Only shows the students with name fully matched.

Actual Behavior: 
Students with name partial matched are all shown.

Steps To Reproduce:
1.	Type “view -s li” and enter

Suggestion: 
1.	Simply update User Guide or 
2.	adjust implementation would solve the problem.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: chuyiting/ped#10

### PR DESCRIPTION
**Pull request by ngzhenteng**
Fix Bug with summary string
Fix view one student bug

Fixes #260

### CODE REVIEW

File: src/test/java/seedu/jarvis/testutil/TypicalAddressBook.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import seedu.jarvis.model.consultation.Consultation;
+import seedu.jarvis.model.masterycheck.MasteryCheck;
+import seedu.jarvis.model.mission.Mission;
+import seedu.jarvis.model.quest.Quest;
+import seedu.jarvis.model.task.Task;
+
+public class TypicalAddressBook {
+
+    public static AddressBook getTypicalAddressBookWithAllValues() {
+        List<Mission> typicalMissions = getTypicalMissions();
```
Comment (wilinetan) [inline src/test/java/seedu/jarvis/testutil/TypicalAddressBook.java]: Perhaps the level of abstraction should be the same? You only imported the getTypicalMissions method and not the rest

File: src/test/java/seedu/jarvis/testutil/TypicalAddressBook.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.List;
+
+import seedu.jarvis.model.AddressBook;
+import seedu.jarvis.model.consultation.Consultation;
+import seedu.jarvis.model.masterycheck.MasteryCheck;
+import seedu.jarvis.model.mission.Mission;
+import seedu.jarvis.model.quest.Quest;
+import seedu.jarvis.model.task.Task;
+
+public class TypicalAddressBook {
```
Comment (wilinetan) [inline src/test/java/seedu/jarvis/testutil/TypicalAddressBook.java:12]: Thanks for adding this class to help with testing!

**Review by wilinetan — COMMENTED** [review]
LGTM thanks for adding more tests

File: src/test/java/seedu/jarvis/testutil/TypicalAddressBook.java (hunk 1 above)
Comment (ngzhenteng) [reply src/test/java/seedu/jarvis/testutil/TypicalAddressBook.java]: Thank you for that eye for detail!

**Pull request merged**